src/main/java/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,40 @@
+package duke.command;
+
+import duke.DukeException;
+import duke.storage.Storage;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Command that deletes a task from the list of tasks.
+ */
+public class DeleteCommand extends Command {
+    private final int index;
+
+    /**
+     * Constructs a DeleteCommand object that deletes a task from the task list upon execution.
+     *
+     * @param index The index of the task to be deleted.
+     */
+    public DeleteCommand(int index) {
+        this.index = index;
+    }
```

Comment:
Could consider using a scratch variable `i` to indicate its short scope and perhaps reduce the use of the `this` keyword?
```suggestion
    public DeleteCommand(int i) {
        index = i;
    }
```